http/swagger.yml

```diff
@@ -8775,24 +8912,26 @@ components:
         - $ref: "#/components/schemas/CheckBase"
         - type: object
           properties:
+            type:
+              type: string
+              enum: [threshold]
             thresholds:
               type: array
               items:
-                oneOf:
-                  - $ref: "#/components/schemas/GreaterThreshold"
-                  - $ref: "#/components/schemas/LesserThreshold"
-                  - $ref: "#/components/schemas/RangeThreshold"
-                discriminator:
-                  propertyName: type
-                  mapping:
-                    greater: "#/components/schemas/GreaterThreshold"
-                    lesser: "#/components/schemas/LesserThreshold"
-                    range: "#/components/schemas/RangeThreshold"
+                $ref: "#/components/schemas/ThresholdCheckThreshold"
+    ThresholdCheckThreshold:
+      oneOf:
+        - $ref: "#/components/schemas/GreaterThreshold"
+        - $ref: "#/components/schemas/LesserThreshold"
+        - $ref: "#/components/schemas/RangeThreshold"
```

Comment:
ThresholdCheckThreshold sounds a little off to me. We would call it ThresholdCheckDetail or ThresholdCheckConfig

For the struct, why don't you make it just one simple object
```go
type ThresholdConfig struct {
	AllValues bool                    `json:"allValues"`
	Level     notification.CheckLevel `json:"level"`
	LowerBound       *float64                `json:"lowerBound,omitempty"`
	UpperBound       *float64                `json:"upperBound,omitempty"`
}
```
So for greater, fill the lowerBound; for lesser fill the upperBound; for range fill both. For within false range, simply reverse the upperBound and lowerBound.

Reply:
I agree it's a weird name. How about just `Threshold`?

My preference would be to make them separate but similar models, so that it is less ambiguous how each variant should be handled. Doing so helps us on the front end, since that style generates a [discriminated union TypeScript definition](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) that provides us type checking benefits (it becomes a compile time error to use the a variant in the the wrong way). 

In any event, I think we would still need a `type` field with value either "lesser", "greater", or "range" in order to distinguish how these models are handled on the front end.